src/ActionMonitor/Monitors/PreviewMonitor.php

```diff
@@ -9,31 +9,593 @@ class PreviewMonitor extends Monitor {
+						'remoteStatus'   => $remote_status_type,
+						'pageNode'       => [
+							'path' => $normalized_preview_page_path,
+						],
+						'modifiedLocal'  => $modified,
+						'modifiedRemote' => $gatsby_node_modified,
+					];
+				},
+			]
+		);
+
+		register_graphql_field(
+			'WPGatsby',
+			'isPreviewFrontendOnline',
+			[
+				'description' => __( 'Wether or not the Preview frontend URL is online.', 'wp-gatsby' ),
+				'type'        => 'Boolean',
+				'resolve'     => function( $root, $args, $context, $info ) {
+					if ( ! is_user_logged_in() ) {
+						return false;
+					}
+
+					$preview_url = self::get_gatsby_preview_instance_url();
+
+					$request = wp_remote_get( $preview_url );
+
+					$request_was_successful =
+						$this->was_request_successful( $request );
+
+					return $request_was_successful;
+				},
+			]
+		);
+	}
+
 	/**
-	 * Get the Gatsby Preview instance refresh webhook
+	 * Determines wether or not the current global 
+	 * PHP context is related to Gatsby Preview
 	 */
-	static function get_gatsby_preview_webhook() {
-		$preview_webhook = Preview::get_setting( 'preview_api_webhook' );
+	public static function is_gatsby_preview() {
+		$is_gatsby_preview 
+			= \is_preview()
+			|| isset( $_GET['preview_nonce'] )
+			|| isset( $_GET['gatsby_preview'] );
 
-		if (
-			! $preview_webhook ||
-			! filter_var( $preview_webhook, FILTER_VALIDATE_URL )
-		) {
-			return false;
+		return $is_gatsby_preview;
+	}
+
+	/**
+	 * If specific conditions are met, this loads the Gatsby Preview template
+	 * instead of the core WordPress preview template
+	 *
+	 * @param string $template The template to load
+	 *
+	 * @return string
+	 */
+	public function setup_preview_template( $template ) {
+		global $post;
+
+		// If the global post isn't set, but the preview_id is passed, use that to determine
+		// the preview post
+		if ( empty( $post ) && isset( $_GET['preview_id'] ) ) {
+			$post = get_post( $_GET['preview_id'] );
 		}
 
-		if ( substr( $preview_webhook, -1 ) !== '/' ) {
-			$preview_webhook .= '/';
+		// Determine the post_type of the post to be previewed
+		$post_type = $post->post_type ?? null;
+
+		// Ensure the post_type is set to show_in_graphql
+		$post_type_object = $post_type ? get_post_type_object( $post->post_type ) : null;
+
+		if ( $post_type && ! $post_type_object->show_in_graphql ?? true ) {
+			return plugin_dir_path( __FILE__ ) . '../../Admin/includes/post-type-not-shown-in-graphql.php';
 		}
 
-		return $preview_webhook;
+		// Determine if it's a preview
+		$is_gatsby_preview = self::is_gatsby_preview();
+		$is_wp_preview = is_preview();
+
+		if ( $is_gatsby_preview && ! $is_wp_preview ) {
+			// WP doesn't call post_save for every second preview with no content changes. Since we're using post_save to trigger the webhook to Gatsby, we need to get WP to call post_save for this post.
+			$update = true;
+			$post_ID = $post->ID;
+			do_action( 'save_post', $post_ID, $post, $update );
```

Comment:
@smthomas we now call the save_post action on every template load for the post being previewed. Inside that action we're creating new events for Gatsby to pick up